cmd/cluster/resizenode.go

```diff
@@ -0,0 +1,342 @@
+package cluster
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/openshift/osdctl/internal/utils/globalflags"
+	k8spkg "github.com/openshift/osdctl/pkg/k8s"
+	awsprovider "github.com/openshift/osdctl/pkg/provider/aws"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// resizeNodeOptions defines the struct for running resizeNode command
+type resizeNodeOptions struct {
+	k8sclusterresourcefactory k8spkg.ClusterResourceFactoryOptions
+	node                      string
+	newMachineType            string
+	verbose                   bool
+
+	genericclioptions.IOStreams
+	GlobalOptions *globalflags.GlobalOptions
+}
+
+// This command requires to be backplane logged into the target cluster
+// During command execution, the OCM context is swapped back and forth: cluster - hive - cluster
+func newCmdResizeNode(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *cobra.Command {
+	ops := newResizeNodeOptions(streams, flags, globalOpts)
+	resizeNodeCmd := &cobra.Command{
+		Use:               "resizeNode",
+		Short:             "Resize a node. Requires previous login to the target cluster via ocm backplane login <target_cluster>",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
```

Comment:
It is a bit strange to me to have a `run` method on options object.
Semantically speaking, it would have preferred having some `run` function taking options as parameter

Reply:
Semantically I tend to agree, however this seems to be a standard in all our cobra cli projects.
See: [oc cli](https://github.com/openshift/oc/blob/c443e1c0b03ee88b44f237d08572346332c9bbe5/pkg/cli/login/login.go#L51) and other commands in `osdctl`